### ISSUE TITLE
Support custom configuration options in accessibility checker

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#10137]
## Is your proposal related to a problem?

Currently, the only supported configuration of the accessibility checker is to turn it on or off like all userbar items. With Axe having so many ways to be configured, it’d make sense to expose some of those options to site implementers.

## Describe the solution you'd like

Have a Wagtail-friendly way to configure:

- Which elements of the page the checker runs on – `main, title` for example (so we don’t report issues on the footer)
- Which elements of the page are ignored – for example `footer, .my-cool-widget`. Making sure we retain our ignoring of `wagtail-userbar`
- Which rules are on/off (if possible including the ability to create fully custom Axe rules) – including customising the error message of each rule if possible.

This can either be an almost direct reproduction of the relevant Axe configuration options, or something more opinionated.

In addition, we could consider supporting configuring the checker differently for:

- Different preview modes
- Different user roles / groups
- Page type by page type

## Describe alternatives you've considered

We could do this via Python/Django, templates, or JS. Python/Django feels much more promising.

---

## Implementation approach

In the interest of time, keep using the userbar hook? Add class attributes to define defaults, methods as relevant. To customise, use the hook to swap default `AccessibilityItem` with the subclass. Retain a way to override whole Axe configuration if possible (doesn’t have to be elegant, or documented).

Documentation: TBC (can be done during code freeze)
